File: src/main/java/com/openpms/server/app/AppController.java
package com.openpms.server.app;

import com.openpms.common.R;
import com.openpms.server.app.dto.LoginBody;
import jakarta.servlet.http.HttpSession;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;
import java.util.Map;

@RequiredArgsConstructor
@RestController
public class AppController {
    private final AppService appService;

    @GetMapping
    public Map<String, Object> index(HttpSession session) {
        Map<String, Object> data = new HashMap<>();
        data.put("name", session.getAttribute("name"));
        return data;
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody LoginBody body) {
        Boolean result = appService.login(body);
        if (!result) {
            return R.WithStatus(0, "登录失败，账户密码不正确或已被禁用", 401);
        }
        return R.Ok("登录成功");
    }
}

File: src/main/kotlin/com/creator/server/app/AppController.kt
package com.creator.server.app

import cn.dev33.satoken.stp.SaTokenInfo
import cn.dev33.satoken.stp.StpUtil
import com.creator.common.R
import com.creator.server.app.dto.LoginBody
import jakarta.servlet.http.HttpSession
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController

@RestController
class AppController {
    private val appService: AppService? = null

    @GetMapping
    fun index(session: HttpSession): Map<String, Any> {
        val data: MutableMap<String, Any> = HashMap()
        data["name"] = session.getAttribute("name")
        return data
    }

    @PostMapping("/login")
    fun login(@RequestBody body: LoginBody): ResponseEntity<*> {
        val result = appService!!.login(body)
        if (!result) {
            return R.WithStatus(0, "登录失败，账户密码不正确或已被禁用", 401)
        }
        val tokenInfo: SaTokenInfo = StpUtil.getTokenInfo()
        return ResponseEntity.ok<SaTokenInfo>(tokenInfo)
    }

    @PostMapping("/logout")
    fun logout(): ResponseEntity<*> {
        StpUtil.logout()
        return R.Ok("登出")
    }
}
